### PR TITLE
(Closes #2206) Ensure that symbol_table deep copy no longer deletes interface renaming

### LINKED ISSUE #2206: PSyclone not handling Use re-name symbols correctly
The `socrates_runes.f90` file contains a `Use` statement that does renaming, and PSyclone is outputting Fortran without these renamings:
This:
```
use rad_pcf, only: &
  ip_source_illuminate                      => ip_solar, &
  ip_source_thermal                         => ip_infra_red, &
  ip_cloud_representation_off               => ip_cloud_off, &
  ip_cloud_representation_ice_water         => ip_cloud_ice_water, &
  ip_cloud_representation_combine_ice_water => ip_cloud_combine_ice_water, &
  ip_cloud_representation_csiw              => ip_cloud_csiw, &
  ip_cloud_representation_split_ice_water   => ip_cloud_split_ice_water, &
  ip_overlap_max_random                     => ip_max_rand, &
  ip_overlap_random                         => ip_rand, &
  ip_overlap_exponential_random             => ip_exponential_rand, &
  ip_inhom_homogeneous                      => ip_homogeneous, &
  ip_inhom_scaling                          => ip_scaling, &
  ip_inhom_mcica                            => ip_mcica, &
  ip_inhom_cairns                           => ip_cairns, &
  ip_inhom_tripleclouds_2019                => ip_tripleclouds_2019, &
  ip_droplet_re_external                    => ip_re_external, &
  ip_droplet_re_constant                    => ip_re_constant, &
  ip_droplet_re_liu                         => ip_re_liu, &
  ip_droplet_re_default                     => ip_re_default
```
is converted to this by PSyclone:
```
  use rad_pcf, only : ip_cloud_representation_combine_ice_water, ip_cloud_representation_csiw, ip_cloud_representation_ice_water, &
&ip_cloud_representation_off, ip_cloud_representation_split_ice_water, ip_droplet_re_constant, ip_droplet_re_default, &
&ip_droplet_re_external, ip_droplet_re_liu, ip_inhom_cairns, ip_inhom_homogeneous, ip_inhom_mcica, ip_inhom_scaling, &
&ip_inhom_tripleclouds_2019, ip_overlap_exponential_random, ip_overlap_max_random, ip_overlap_random, ip_source_illuminate, &
&ip_source_thermal
```

This then fails to compile as the compiler looks in the module for these variables but fails to find them.

### PR DESCRIPTION
**Pull request by LonelyCat124**
Symbol table was not correctly handling interface renaming during deep copy. This fixes that error and adds a test to show that behaviour is correct now.

### CODE REVIEW

File: src/psyclone/psyir/symbols/symbol_table.py [hunk 1]
```diff
@@ -282,8 +283,10 @@ def deep_copy(self):
         # Fix the container links for imported symbols
         for symbol in new_st.imported_symbols:
             name = symbol.interface.container_symbol.name
+            orig_name = symbol.interface.orig_name
```
Comment (arporter) [inline src/psyclone/psyir/symbols/symbol_table.py:286]: Please could you add/extend a test of the deep-copy functionality in `symbol_table_test.py`?

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good, thanks for finding this and fixing it.
As it's a bug fix there's no need to update any docs.
Changes are pylint clean.
Just a request to extend the testing of the deep-copy method itself and then this can go on.

File: src/psyclone/psyir/symbols/symbol_table.py (hunk 1 above)
Comment (LonelyCat124) [reply src/psyclone/psyir/symbols/symbol_table.py:286]: Added it to the existing `test_deep_copy` test.

**Comment by LonelyCat124** [comment]
Test added, should be good to go now.

**Pull request merged**